tests/test/scheduler/test_function_client_server.cpp

```diff
@@ -53,32 +53,51 @@ class ClientServerFixture
 
 TEST_CASE_METHOD(ClientServerFixture, "Test sending MPI message", "[scheduler]")
 {
-    // Create an MPI world on this host and one on a "remote" host
-    std::string otherHost = "192.168.9.2";
+    auto& sch = faabric::scheduler::getScheduler();
```

Comment:
Removing `registerRank` and having everything be handled by the bootstrapping process has the downside that we now need to do a better job at faking the MPI environment in tests to ensure the expected rank allocation.

Reply:
Yes this is a bit of a hassle but doesn't look _too_ bad. Ideally we'd move these sort of tests to "proper" distributed tests, but this will do for now.